From 22ce9aa7c43e574653a8b22e180fc835e093aaa5 Mon Sep 17 00:00:00 2001
From: De Huo <De.Huo@windriver.com>
Date: Thu, 25 Jul 2019 17:26:22 +0800
Subject: [PATCH] Add WRLinux1019 test case for rule
 cracklib_accounts_password_pam_retry

Add WRLinux1019 specific test case cracklib_accounts_password_pam_retry.

Upstream-Status: Pending

Signed-off-by: De Huo <De.Huo@windriver.com>
---
 .../bash/wrlinux1019.sh                            | 13 +++++++++
 .../oval/wrlinux1019.xml                           | 32 ++++++++++++++++++++++
 2 files changed, 45 insertions(+)
 create mode 100644 linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_retry/bash/wrlinux1019.sh
 create mode 100644 linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_retry/oval/wrlinux1019.xml

diff --git a/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_retry/bash/wrlinux1019.sh b/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_retry/bash/wrlinux1019.sh
new file mode 100644
index 0000000..c8f9785
--- /dev/null
+++ b/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_retry/bash/wrlinux1019.sh
@@ -0,0 +1,13 @@
+# platform = multi_platform_wrlinux
+. /usr/share/scap-security-guide/remediation_functions
+populate var_password_pam_retry
+
+if grep -q '^[^#].*pam_cracklib.so' /etc/pam.d/common-password; then
+    if grep -q "retry=" /etc/pam.d/common-password ; then
+    	sed -i --follow-symlinks "s/\(^password.*pam_unix.so.*\)\(retry *= *\).*/\1retry=$var_password_pam_retry/" /etc/pam.d/common-password
+    else    
+    	sed -i --follow-symlinks "/^password[[:space:]]\+.*\+pam_cracklib.so/ s/$/ retry=$var_password_pam_retry/" /etc/pam.d/common-password
+    fi
+else
+    echo "password required pam_cracklib.so retry=$var_password_pam_retry">>/etc/pam.d/common-password
+fi
diff --git a/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_retry/oval/wrlinux1019.xml b/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_retry/oval/wrlinux1019.xml
new file mode 100644
index 0000000..3854001
--- /dev/null
+++ b/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_retry/oval/wrlinux1019.xml
@@ -0,0 +1,32 @@
+<def-group>
+  <definition class="compliance" id="cracklib_accounts_password_pam_retry" version="1">
+    <metadata>
+      <title>Set Password retry Requirements</title>
+      <affected family="unix">
+        <platform>multi_platform_wrlinux</platform>
+      </affected>
+      <description>The password retry should meet minimum
+      requirements</description>
+    </metadata>
+    <criteria>
+      <criterion comment="wrlinux1019 pam_cracklib" test_ref="test_password_pam_cracklib_retry" />
+    </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test check="all" comment="check the configuration of /etc/pam.d/system-auth" id="test_password_pam_cracklib_retry" version="1">
+    <ind:object object_ref="obj_password_pam_cracklib_retry" />
+    <ind:state state_ref="state_password_pam_cracklib_retry" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="obj_password_pam_cracklib_retry" version="1">
+    <ind:filepath>/etc/pam.d/common-password</ind:filepath>
+    <ind:pattern operation="pattern match">^\s*password\s+(?:(?:required)|(?:requisite))\s+pam_cracklib\.so.*retry=([0-9]*).*$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_state id="state_password_pam_cracklib_retry" version="1">
+    <ind:subexpression datatype="int" operation="less than or equal" var_ref="var_password_pam_retry" />
+  </ind:textfilecontent54_state>
+
+  <external_variable comment="External variable for pam_cracklib retry" datatype="int" id="var_password_pam_retry" version="1" />
+</def-group>
-- 
1.9.1

